lib/index.d.ts

```diff
@@ -0,0 +1,195 @@
+
+  /**
+   * Deletes an entity and all its related components and relationships.
+   * @param id Entity to be destroyed
+   */
+  delete(id: Entity): void;
+
+  /**
+   * Adds a component to the entity with no value
+   * @param id Target Entity
+   * @param component Component
+   */
+  add<T>(id: Entity, component: Entity<T>): void;
+
+  /**
+   * Assigns a value to a component on the given entity
+   * @param id Target Entity
+   * @param component Target Component
+   * @param data Component Data
+   */
+  set<T>(id: Entity, component: Entity<T>, data: T): void;
+
+  /**
+   * Removes a component from the given entity
+   * @param id Target Entity
+   * @param component Target Component
+   */
+  remove(id: Entity, component: Entity): void;
+
+  // Manually typed out get since there is a hard limit.
+
+  /**
+   * Retrieves the value of one component. This value may be undefined.
+   * @param id Target Entity
+   * @param component Target Component
+   * @returns Data associated with the component if it exists
+   */
+  get<A>(id: number, component: Entity<A>): A | undefined;
+
+  /**
+   * Retrieves the value of two components. This value may be undefined.
+   * @param id Target Entity
+   * @param component Target Component 1
+   * @param component2 Target Component 2
+   * @returns Data associated with the components if it exists
+   */
+  get<A, B>(
+    id: number,
+    component: Entity<A>,
+    component2: Entity<B>
+  ): LuaTuple<Nullable<[A, B]>>;
+
+  /**
+   * Retrieves the value of three components. This value may be undefined.
+   * @param id Target Entity
+   * @param component Target Component 1
+   * @param component2 Target Component 2
+   * @param component3 Target Component 3
+   * @returns Data associated with the components if it exists
+   */
+  get<A, B, C>(
+    id: number,
+    component: Entity<A>,
+    component2: Entity<B>,
+    component3: Entity<C>
+  ): LuaTuple<Nullable<[A, B, C]>>;
+
+  /**
+   * Retrieves the value of four components. This value may be undefined.
+   * @param id Target Entity
+   * @param component Target Component 1
+   * @param component2 Target Component 2
+   * @param component3 Target Component 3
+   * @param component4 Target Component 4
+   * @returns Data associated with the components if it exists
+   */
+  get<A, B, C, D>(
+    id: number,
+    component: Entity<A>,
+    component2: Entity<B>,
+    component3: Entity<C>,
+    component4: Entity<D>
+  ): LuaTuple<Nullable<[A, B, C, D]>>;
+
+  /**
+   * Searches the world for entities that match a given query
+   * @param components Queried Components
+   * @returns Iterable function
+   */
+  query<T extends Entity[]>(...components: T): Query<InferComponents<T>>;
+}
+
+export const pair: (pred: Entity, obj: Entity) => Entity;
+
+export const OnAdd: Entity;
+export const OnRemove: Entity;
+export const OnSet: Entity;
+export const Wildcard: Entity;
+export const w: Entity;
+export const REST: Entity;

```

Comment:
You can remove these types since they won't be used as apart of the user facing API anyways, and with that you can also remove the entity index type and the other related types

Reply:
New exports currently match [this API](https://github.com/Ukendio/jecs/blob/main/docs/api/world.md)